### PR TITLE
Magic bundle: Added thing that schedules a job to go ONLINE after a delay

### PR DESCRIPTION
**Pull request by triller-telekom**
Signed-off-by: Stefan Triller <stefan.triller@telekom.de>

### CODE REVIEW

**Comment by triller-telekom** [comment]
I have created this addition to the magic bundle in order to track down https://github.com/eclipse/smarthome/issues/4931 where we had to create the revert PR https://github.com/eclipse/smarthome/pull/4772 for the [desperately wanted feature](https://github.com/openhab/openhab2-addons/pull/2859#issuecomment-363222559) from PR: https://github.com/eclipse/smarthome/pull/4932

File: bundles/test/org.eclipse.smarthome.magic/ESH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -73,4 +73,12 @@
 		</config-description>
 	</thing-type>
 
+	<thing-type id="delayed-thing">
+        <label>Magic Delayed Online Thing</label>
+        <description>A thing that goes online after some time.</description>
+
+        <channels>
+            <channel id="switch" typeId="switch" />
```
Comment (maggu2810) [inline bundles/test/org.eclipse.smarthome.magic/ESH-INF/thing/thing-types.xml]: You could use a number and the implementation could use that number of seconds to delay the go online state update. A negative number could schedule a go offline state update. (Just an idea)

Comment (htreu) [reply bundles/test/org.eclipse.smarthome.magic/ESH-INF/thing/thing-types.xml]: Yes, nice. And for a new command to this channel the Thing will go OFFLINE, wait the amount of seconds from the command (if positive) and go ONLINE again.

Comment (maggu2810) [reply bundles/test/org.eclipse.smarthome.magic/ESH-INF/thing/thing-types.xml]: @htreu Okay for me, too. But my suggestion has been a different one :wink: 
My suggestion has been:
* Receiving a command does not change the state immediately.
* If the received number is > 0, then an online state update will be scheduled with a delay of number.
* If the received number is < 0, then an offline state update will be scheduled with a delay of -number.

So, in short:
* delay = Math.abs(number)
* state = number > 0 ? ONLINE : OFFLINE

Comment (triller-telekom) [reply bundles/test/org.eclipse.smarthome.magic/ESH-INF/thing/thing-types.xml]: Good idea, I have changed the code to reflect that and as a bonus I can reproduce #4931 with that!

**Pull request merged**